### ISSUE TITLE
Prettier eslint integration is not working

### ISSUE THREAD

**Issue by bsvipas** [prettier/prettier-vscode#189]
Hello,

the `"prettier.eslintIntegration": true`is not working. I tried to install `prettier-eslint` as local and global package and it was the same. I tried to use `eslint` as local and as global package and it was the same. I even tried to install all three packages including `prettier` to local and global and it was the same. I tried different variations and it wasn't working. I mean it's formatting, but it doesn't do `eslint --fix`.

The Prettier output is clean, it has in sidebar ✓ symbol. ESLint output is good too.

VS Code: 1.14.2
macOS Sierra: 10.12.6 (16G29)

**Comment by RobinMalfait** [comment]
Could you share your eslint file and an example input file and the end result?

**Comment by bsvipas** [comment]
Code:
```javascript
const aFn = (data) => console.log(data);
```

Prettier formats to:
```javascript
const aFn = data => console.log(data);
```

ESLint:
```json
{
	"rules": {
		"arrow-parens": "error"
	}
}

```

**Comment by RobinMalfait** [comment]
I don't know the `json` representation for this, but when I use the `yaml` syntax it works with this:

```yaml
---
  env:
    es6: true
  rules:
    arrow-parens:
      - 2
      - always
```

**Comment by RobinMalfait** [comment]
Note that the `es6` env is required because of the `const` and `arrow function` syntax.

**Comment by bsvipas** [comment]
I have `es6`.

```json
	"env": {
		"browser": true,
		"commonjs": true,
		"es6": true,
		"node": true
	},
```

**Comment by CiGit** [comment]
What @RobinMalfait meant was, in `json` form
`"arrow-parens": ["error", "always"]`
http://eslint.org/docs/rules/arrow-parens
Your `arrow-parens` seems wrongly configured.

**Comment by RobinMalfait** [comment]
Yes indeed, like @CiGit said. This works for me:

```json
{
  "env": {
    "es6": true
  },
  "rules": {
    "arrow-parens": ["error", "always"]
  }
}
```

Can you confirm that this solves your issue, @bsvipas ?

**Comment by bsvipas** [comment]
@CiGit @RobinMalfait It's configured good, ESLint works in both ways if you left only `"error"` or `["error", "always"]`. I tried to use `["error", "always"]` and it wasn't working (I'm talking about Prettier). Using one param for e.g. `"warn"` or `"error"` always works if that rule have default type in this example `"arrow-parens"` default type is **always**. I can use both approaches `"error"` or `["error", "always"]` and ESLint always works without any problems.

I tried on new and clean project with only `index.js` and `.eslintrc.json`:

```json
{
	"env": {
		"es6": true
	},
	"rules": {
		"arrow-parens": "error"
	}
}
```

And it was working! Prettier doesn't remove the parens, BUT how to fix that on my project? I think maybe it can be, because that project is having 2 ESLint files and both are not in root, the structure is like this:

root ->
- api -> 
  - .eslintrc.json
  - other files
- frontend ->
  - .eslintrc.json`
  - other files
- other files, but no .eslintrc.json in root

I configured my VS Code workspaces settings to use ESLint config from frontend folder and it's working for ESLint, but Prettier is not working, I even tried to add .eslintrc.json in the root, but Prettier still is not working... Maybe we need new prettier setting to show where is and which ESLint config to use? Or Prettier is communicating with ESLint to know correct place of config?

In short: Prettier is working if everything is in root, but if I have different structure it doesn't working, but ESLint does. I mean it's formatting, but not respecting ESLint configuration even if integration is true.

**Comment by CiGit** [comment]
We don't communicate with eslint extension. We use `prettier-eslint`. And due to how it works, there is no root folder for eslint.
Have you installed eslint in your project?

**Comment by bsvipas** [comment]
@CiGit Yes, I installed it. As I said I tried to do everything, install it global, local, but nothing helps.

**Comment by CiGit** [comment]
Globally should help much. We don't use it.
How did you install it locally ? 
Was it installed in your test repo ? Versions?

**Comment by lucasfeliciano** [comment]
Same thing here:

Prettier is running but prettier-eslint it isn't 
I have `prettier-eslint` installed locally

**Comment by lucasfeliciano** [comment]
I got the problem:

`"prettier.eslintIntegration": true` 

Need to be set in the editor config file.
**Not** on `.prettierrc`/`package.json`/`.prettier.config.js`

**Comment by bsvipas** [comment]
Seems everything fine for me, I guess I can close this issue.

**Comment by yinsang** [comment]
@lucasfeliciano  I set `"prettier.eslintIntegration": true`in vscode setting ，then it works.

**Comment by kirkbross** [comment]
"prettier.tslintIntegration" and "prettier.eslintIntegration" in VSCode both give me "Unknown Configuration Setting" error.
